lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1 # nothing has been set yet so both will be set to -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if ((@back + 1) % @store.length) == @front
+      raise StandardError.new
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length # but what happens when it reaches the end?
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    element = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+    return element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    if @front < @back
+      return @back - @front
+    else
+      return @front - @back
+    end
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return true if @front == @back
+    return false
   end
 
   def to_s
+    # @front = (@front) % @store.length
+    # @back = (@back) % @store.length
+    if @front < @back
+      @store = @store[@front...@back]
+    else
+      @store = @store[@back...@front]
```

Comment:
This won't quite work, as it will get the blank cells between back and front.  You need to get from front to the end of the array and from the front of the array to back.  **Also don't change store!**

```suggestion
     return "#{@store[(@back + 1).. -1] + @store[0..@front]}"
```